provider/aws/aws.go

```diff
@@ -788,18 +783,29 @@ func (p *AWSProvider) AdjustEndpoints(endpoints []*endpoint.Endpoint) ([]*endpoi
 			} else {
 				ep.SetProviderSpecificProperty(providerSpecificEvaluateTargetHealth, strconv.FormatBool(p.evaluateTargetHealth))
 			}
+
+			if ep.RecordType == endpoint.RecordTypeCNAME {
```

Comment:
is it not safer to create a new Endpoint? To separate the creation of the AAAA record and the modification of the original record type.

```
if ep.RecordType == endpoint.RecordTypeCNAME {
	// This needs to match two records from Route53, one alias for 'A' (IPv4)
	// and one alias for 'AAAA' (IPv6).
	aliasCnameAaaaEndpoints = append(aliasCnameAaaaEndpoints, &endpoint.Endpoint{
		DNSName:    ep.DNSName,
		Targets:    ep.Targets,
		RecordType: endpoint.RecordTypeAAAA,
		RecordTTL:  ep.RecordTTL,
		Labels:     ep.Labels,
		ProviderSpecific: ep.ProviderSpecific,
		SetIdentifier: ep.SetIdentifier,
	})
	ep.RecordType = endpoint.RecordTypeA
}
```

Reply:
I'll be first to admit that whilst I enjoy Go I rarely get to use it day to day. What would the advantage be in creating a new endpoint as opposed to cloning the existing one?

Reply:
So we not accidently change values, as you passing it by reference in current implementation

Reply:
I see, that could be a misunderstanding on my part. If there is a way of cloning it (without just referencing it) would that not be better? Since if the structure of "endpoint.Endpoint" ever changes, we need to update the fields in this fairly obscure place also?

Reply:
I shared an example that could be described as cloning.

Reply:
I thought that:
```
				epCnameAaaa := *ep
```

That the `*` here takes the value of the pointer, therefore it would be safe and saves us needing to know the full struct details of `endpoint`. That said, I am happy to accept I've misunderstood and I will make the changes you have suggested after just confirming this point.

Reply:
and `type Targets []string` is a slice, so worth to validate the bahaviour.

Reply:
You raise a good point about slices and maps, I will implement and retest it with your suggestion.